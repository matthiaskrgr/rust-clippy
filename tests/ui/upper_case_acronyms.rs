#![warn(clippy::upper_case_acronyms)]
#![allow(clippy::fully_upper_case_idents)]

struct HTTPResponse; // linted

struct CString; // not linted

enum Flags {
    NS, // linted
    CWR,
    ECE,
    URG,
    ACK,
    PSH,
    RST,
    SYN,
    FIN,
}

struct GCCLLVMSomething; // linted, beware that lint suggests `GccllvmSomething` instead of `GccLlvmSomething`

fn main() {}
