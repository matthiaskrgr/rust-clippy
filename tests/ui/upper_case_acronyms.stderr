error: name `HTTPResponse` contains a capitalized acronym
  --> $DIR/upper_case_acronyms.rs:4:8
   |
LL | struct HTTPResponse; // linted
   |        ^^^^^^^^^^^^ help: consider making the acronym lowercase, except the initial letter: `HttpResponse`
   |
   = note: `-D clippy::upper-case-acronyms` implied by `-D warnings`

error: name `NS` contains a capitalized acronym
  --> $DIR/upper_case_acronyms.rs:9:5
   |
LL |     NS, // linted
   |     ^^ help: consider making the acronym lowercase, except the initial letter (notice the capitalization): `Ns`

error: name `GCCLLVMSomething` contains a capitalized acronym
  --> $DIR/upper_case_acronyms.rs:20:8
   |
LL | struct GCCLLVMSomething; // linted, beware that lint suggests `GccllvmSomething` instead of `GccLlvmSomething`
   |        ^^^^^^^^^^^^^^^^ help: consider making the acronym lowercase, except the initial letter: `GccllvmSomething`

error: aborting due to 3 previous errors

